[WAF] CrowdSec crashes if it cannot bind AppSec Port (7422)

### What happened?

If port 7422 (or whatever is used by appsec acquis) is already used, appsec/crowdsec will crash on startup

```
INFO[2025-09-11T13:46:32+02:00] Appsec Runner is dying                        name=myAppSecComponent runner_uuid=b357bb99-51e5-41b0-8fc4-71847de01d78 type=appsec
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x7adf67]

goroutine 135 [running]:
net/http.(*onceCloseListener).close(...)
	net/http/server.go:3969
sync.(*Once).doSlow(0xc0011fccb8?, 0xc000fbdbe0?)
	sync/once.go:78 +0xac
sync.(*Once).Do(...)
	sync/once.go:69
net/http.(*onceCloseListener).Close(0xc00128a240)
	net/http/server.go:3965 +0x3b
panic({0x1fcbe20?, 0x3952b00?})
	runtime/panic.go:783 +0x132
net/http.(*onceCloseListener).Accept(0x27d0478?)
	<autogenerated>:1 +0x1e
net/http.(*Server).Serve(0xc0011fcc00, {0x0, 0x0})
	net/http/server.go:3463 +0x30c
github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/appsec.(*AppsecSource).listenAndServe.func1({0x0?, 0x0?}, 0xba?)
	github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/appsec/appsec.go:366 +0x116
github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/appsec.(*AppsecSource).listenAndServe.func3({0xc0014642e8, 0xe})
	github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/appsec/appsec.go:413 +0x1c2
created by github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/appsec.(*AppsecSource).listenAndServe in goroutine 132
	github.com/crowdsecurity/crowdsec/pkg/acquisition/modules/appsec/appsec.go:402 +0x314
```

### What did you expect to happen?

graceful fail

### How can we reproduce it (as minimally and precisely as possible)?

n/a

### Anything else we need to know?

n/a

### Crowdsec version

<details>

```console
$ cscli version
# paste output here
```

</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>


### Enabled collections and parsers

<details>

```console
$ cscli hub list -o raw
# paste output here
```

</details>


### Acquisition config

<details>
```console
# On Linux:
$ cat /etc/crowdsec/acquis.yaml /etc/crowdsec/acquis.d/*
# paste output here

# On Windows:
C:\> Get-Content C:\ProgramData\CrowdSec\config\acquis.yaml
# paste output here
</details>


### Config show

<details>

```console
$ cscli config show
# paste output here
```

</details>


### Prometheus metrics

<details>

```console
$ cscli metrics
# paste output here
```

</details>


### Related custom configs versions (if applicable) : notification plugins, custom scenarios, parsers etc.

<details>

</details>

--- a/pkg/acquisition/modules/appsec/appsec.go
+++ b/pkg/acquisition/modules/appsec/appsec.go
@@ -407,6 +407,7 @@ func (w *AppsecSource) listenAndServe(ctx context.Context, t *tomb.Tomb) error {
 		listener, err := net.Listen("tcp", url)
 		if err != nil {
 			serverError <- fmt.Errorf("listening on %s: %w", url, err)
+			return
 		}
 
 		w.logger.Infof("Appsec listening on %s", url)

